src/ducks/apps/index.js

```diff
@@ -149,25 +149,32 @@ function _sortAlphabetically (array, property) {
   return array.sort((a, b) => a[property] > b[property])
 }
 
+/* Only for the icon fetching */
+const root = document.querySelector('[role=application]')
+const data = root && root.dataset
+const COZY_TOKEN = data && data.cozyToken
+const COZY_DOMAIN = data && (`//${data.cozyDomain}`)
```

Comment:
Is there any other way to avoid dealing with an access to `document` here ? I think that it should be better to instantiate and initialize everything that deal with the `root.dataset` directly in the `index.ejs` file and never hear of it again in the part of the code.

Also, we could pass these data in a prop `appData` and use it as a parameter for `_getIcon(url, appData={})`.

Last option, pass it into react `context` with a top component wrapping everything. For example, the `App` component. See https://medium.com/differential/react-context-and-component-coupling-86e535e2d599.
What do you think ?

Reply:
Clearly I don't like this usage. But this is particularly due to a cozy-client-js lack, I have to get that data from myself here (or whenever before in the components tree) just because we don't have a `cozy.client.fetch` which just do a `fetch` by adding the token and the domain correctly as we have for `cozy.client.fetchJSON`.

Reply:
Could be fixed by the next CozyClient, I added a PR https://github.com/cozy/cozy-client/pull/21
I prefer to keep all here instead of change the application outline just to handle this two "temporary" informations.

Reply:
Let's wait for CozyClient.